lib/game.rb

```diff
@@ -1,29 +1,23 @@
 module TicTacToe
 
   class Game 
-    TTT_MARKER1 = "X"
-    TTT_MARKER2 = "O"
-    
-    def initialize(game_turn, marker_loop, game_completion, game_eval)
+    attr_reader :player1
+    attr_reader :player2
+
+    def initialize(game_turn, game_eval, p1_marker, p2_marker)
       @game_turn = game_turn
-      @marker_loop = marker_loop
-      @game_completion = game_completion
       @game_eval = game_eval
+      @player1 = Player.new(1, p1_marker)
+      @player2 = Player.new(2, p2_marker)
     end
 
-    def run 
-      marker1, marker2 = @marker_loop.marker_selection(TTT_MARKER1, TTT_MARKER2)
-
-      @player1, @player2 = Player.new(1, marker1), Player.new(2, marker2)
-
+    def players_turns
       current_player = @player1
-
       begin 
         board_array = @game_turn.execute(current_player.ord_num, current_player.marker)
         current_player = toggle_player(current_player)
-      end until @game_eval.game_over?(board_array, TTT_MARKER1, TTT_MARKER2)
-      
-      @game_completion.game_over_messages(board_array, @player1, @player2)
+      end until @game_eval.game_over?(board_array, @player1, @player2)
```

Comment:
What do you think about only passing the player mark into this method? Your previous implementation was doing this, but you were feeding in the constant values which is an issue if the player marks deviate from the traditional 'X' and 'O'. I don't see anything wrong with your new implementation, but I did notice that now your GameEval class is dependent upon the Player object interface (i.e. it knows that it is expecting an object that responds to `#marker`). I think we have reason to be skeptical of change pertaining to Player since 1) you weren't sure about how useful this object would continue to be and 2) you will be introducing new changes pertaining to different types of players in the near future. So, by changing line 19 to `end until @game_eval.game_over?(board_array, @player1.marker, @player2.marker)` you would be achieving flexibility in terms of player marker values, but without exposing the Player abstraction to your GameEval, thus isolating your GameEval from the possibility of future change to Player.